api/Controllers/HotelsController.cs

```diff
@@ -74,5 +97,69 @@ public async Task<IActionResult> GetHotelCards([FromQuery] PageParameters pagePa
         [HttpGet]
         public async Task<IActionResult> GetHotelsCount() =>
             Ok(await _hotelsDb.Hotels.CountAsync());
+
+        [Route("{hotelId}/images")]
+        [HttpPost]
+        [Authorize(Roles = Role.Admin)]
+        public async Task<IActionResult> AddImage([FromRoute] int hotelId)
+        {
+            try
+            {
+                var file = Request.Form.Files[0];
+                if (file.Length <= 0)
+                {
+                    return BadRequest("Something is wrong with file, probably file is empty");
+                }
+                string fileName = Guid.NewGuid() + Path.GetExtension(file.FileName);
+                string fullPath = Path.Combine(_fileFolder, fileName);
+                using (var stream = new FileStream(fullPath, FileMode.Create))
+                {
+                    await file.CopyToAsync(stream);
+                }
+                ImageEntity image = new ImageEntity
+                {
+                    Path = fileName,
+                    HotelId = hotelId,
+                    Hotel = await _hotelsDb.Hotels
+                        .Where(h => h.Id == hotelId)
+                        .SingleOrDefaultAsync(),
+                    IsOuterLink = false
+                };
+                await _hotelsDb.Images.AddAsync(image);
+                await _hotelsDb.SaveChangesAsync();
+                return CreatedAtAction(nameof(GetImage), new { hotelId = image.Hotel.Id, imageId = image.Id }, null);
+            }
+            catch (Exception ex)
+            {
+                return BadRequest(ex);
+            }
+        }
+
+        [Route("{hotelId}/images")]
+        [HttpGet]
+        public async Task<IActionResult> GetImagesId([FromRoute] int hotelId)
+        {
+            if (await _hotelsDb.Hotels.SingleOrDefaultAsync( h => h.Id == hotelId) == null)
```

Comment:
It is better to use FirstOrDefault for database queries when you are certain that there will be only one record (e.g. getting something by primary key or unique column) or if you actually don't care if there are many records.

Reply:
Yes, and I understand why, we have discussed it on t-sql check